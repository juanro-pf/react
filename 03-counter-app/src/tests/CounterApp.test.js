import '@testing-library/jest-dom';
import React, { Children } from 'react';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas al CounterApp', () => {

    let wrapper= shallow(<CounterApp value={ 100 }/>);

    beforeEach(() => {

        wrapper= shallow(<CounterApp value={ 100 }/>);

    })
    
    test('CounterApp snapshot ', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('CounterApp value ', () => {
        
        // const wrapper= shallow(<CounterApp value={ 100 }/>);

        const value= wrapper.find('h2').at(0).text();

        expect(Number(value)).toBe(100);

    });

    test('+1', () => {

        wrapper.find('button').at(0).simulate('click');

        const counter= wrapper.find('h2').at(1).text();

        expect(Number(counter)).toBe(101);
        
    });

    test('-1', () => {

        wrapper.find('button').at(2).simulate('click');

        const counter= wrapper.find('h2').at(1).text();

        expect(Number(counter)).toBe(99);
        
    });

    test('reset', () => {
        
        wrapper.find('button').at(0).simulate('click');
        console.log(wrapper.find('h2').at(1).text())
        wrapper.find('button').at(0).simulate('click');
        console.log(wrapper.find('h2').at(1).text())
        wrapper.find('button').at(1).simulate('click');
        console.log(wrapper.find('h2').at(1).text())

        const counter= wrapper.find('h2').at(1).text();

        expect(Number(counter)).toBe(100);

    })
    
    
});
