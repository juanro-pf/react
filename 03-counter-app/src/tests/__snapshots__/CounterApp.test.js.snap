// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al CounterApp CounterApp snapshot  1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
